### ISSUE TITLE
This member must have an 'override' modifier because it overrides a member in the base class 'Error'.

### ISSUE THREAD

**Issue by ranouf** [RicoSuter/NSwag#4006]
Hi,

When I generate the Typescript Api Services from a C# API, I have an issue with Exception class generated by NswagStudio:
```
export class ApiException extends Error {
    message: string; //here is the error
    status: number;
    response: string;
    headers: { [key: string]: any; };
    result: any;

    constructor(message: string, status: number, response: string, headers: { [key: string]: any; }, result: any) {
        super();

        this.message = message;
        this.status = status;
        this.response = response;
        this.headers = headers;
        this.result = result;
    }

    protected isApiException = true;

    static isApiException(obj: any): obj is ApiException {
        return obj.isApiException === true;
    }
}
```
Here is my configuration in NswagStudio for the exception 
![image](https://user-images.githubusercontent.com/1764519/168106569-1349a42e-f8a3-4c71-9fe8-777555eeeb99.png)
Version:
![image](https://user-images.githubusercontent.com/1764519/168106882-65c0e090-f186-4201-85f6-321d50f9e721.png)

And the error :

> Error: src/app/services/api/api.services.ts:1915:5 - error TS4114: This member must have an 'override' modifier because it overrides a member in the base class 'Error'.
> 
> 1915     message: string;
>          ~~~~~~~

I can't see any option about that in nswag studio. As the file is automatically generated, I would like to avoid to modifiy the typescript api services files each time the file is updated.

Is there a specific option to fix the issue?

Thanks

**Comment by lahma** [comment]
What's your TS version in nswag.json?

**Comment by lilletech** [comment]
The issue is already known since novembre 2021.
=> https://github.com/RicoSuter/NSwag/issues/3685

Here, many people are waiting for it resolution.

**Comment by ovska** [comment]
You need to use `"typeScriptVersion": 4.3` in your nswag configuration for the `override` keyword to be generated.

**Comment by lilletech** [comment]
@ovska which version do you use ?
<img width="1898" alt="image" src="https://user-images.githubusercontent.com/7681445/168263290-371ff418-ae8c-47a9-aaa5-cc0b7fbc31e0.png">

**Comment by ovska** [comment]
I have the same version, and 4.3 shows up on the dropdown (don't mind the Win95 look; I'm using it through Wine).
![image](https://user-images.githubusercontent.com/68028366/168286071-cc3427ca-8d57-45f2-9e3a-407a5bce9ab3.png)
![image](https://user-images.githubusercontent.com/68028366/168286359-9f1d0831-0a9e-40b4-b971-02593a073356.png)

Try downloading the latest release as zip, and using the NSwagStudio.exe inside. I've had some problems with the msi builds lately.

**Comment by lilletech** [comment]
@ovska yes, portable version shows the 4.3 typescript version. (Not the msi).

**Comment by sander1095** [comment]
@lilletech  @ranouf 

Hi! Our `nswag.json` also uses `4.3` as the typescript version, but `override` is still not generated for the message.

Does your generated typescript client include the `override` keyword? If so, could you post your `nswag.json` so I can compare?

**Comment by joerygrolleman** [comment]
Hi @sander1095 , we managed to fix this issue by upgrading the NSwag packages from version 13.14.8 to 13.15.10.

**Comment by TomONeill** [comment]
Yeah the `override` is being added for errors as seen here:
https://github.com/RicoSuter/NSwag/blob/4749251c712d90ff906251e23f8044031993c934/src/NSwag.CodeGeneration.TypeScript/Templates/File.liquid#L129

but it's not being added for classes that extend from base classes. At least not in version 13.17.0.

**Comment by rwb196884** [comment]
This is still broken.

What is the fix? (My `pakcage.json` says `"tslib": "^2.3.0"` so all this talk about a version `4` seems odd.)

**Comment by BenRacicot** [comment]
In my travels I've landed here thought I'd help out if possible...

I noticed that `public` does not cause TS to give this error. `private` was.

**Comment by Gyorgy-Szy** [comment]
Same issue here. Client was fine, but after updated swagger to 13.20.0, override is missing. For me, it is in lots of places. :(
Any ideas?

**Comment by maxbechtold** [comment]
Same issue as @ranouf 
```
nswag@npm:14.0.3
"typeScriptVersion": 5.0
```

**Comment by kratofl** [comment]
I've got the same issue with the latest versions.
But there's a workaround.

In your tsconfig.json you can turn off this setting:
`"noImplicitOverride": false,`

then it will work.
